jvector-base/src/main/java/io/github/jbellis/jvector/quantization/PQLayout.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.jbellis.jvector.quantization;
+
+/**
+ * Chunk Dimensions and Layout
+ * This is emulative of modern Java records, but keeps to J11 standards.
+ * This class consolidates the layout calculations for PQ data into one place
+ */
+public class PQLayout {
+
+    /** total number of vectors **/
+    public final int vectorCount;
```

Comment:
vectorCount and compressedDimension are stored twice, in PQVectors and here. Should we drop one of them for clarity? Maybe instantiate the layout as
`PQLayout layout = new PQLayout(in.readInt(), in.readInt());`
and we drop the variables in PQVectors

Reply:
I think this section of code is due for some other refactorings. I'd rather avoid putting buffer-mutating calls into an argument position in this particular case.

As another change, we should probably consider making this initialization logic a little less ad-hoc with respect to raw data and buffer pointers.